[Nihirraa] ip

Nihirraa opened the pull request:
# Espresso☕
> "I'm working late...coz I'm a coder" 

## Espresso is your task management chatbot that will help you:

• stay organized
• stay productive
• stay updated

### Here's how you can use EspressoBot:

1. Download it from [here](https://github.com/Nihirraa/ip).
2. Open the application.
3. Start adding your tasks and deadlines!
4. Let Espresso do the rest for you! ✨

> "Isn't that sweet? I guess so"

### Key Features:
-  Managing tasks
-  Managing deadlines (coming soon)
- Notifications and reminders (coming soon)

## **_To-Do List_**
- [ ] Buy Textbooks
- [ ] Organise shelf
- [x] Attend tp meeting
- [x] Buy Grocery

If you're familiar with Java, you can dive into the code and explore. Here's a snippet of the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Nihirraa committed:
Rename main class and add greeting - Level-0

Nihirraa committed:
Level1

Nihirraa committed:
Level2

Nihirraa committed:
Level3

Nihirraa committed:
Level 4

Nihirraa committed:
Added Automated Testing

Nihirraa committed:
Level5

Nihirraa committed:
Level6

tsulemornorfi commented on src/main/java/todoTask.java:
```diff
@@ -0,0 +1,12 @@
+public class todoTask extends Task{
```
class name should start with capital letter

currynia reviewed (commented):
Class names should start with capital letter

ChinSekYi commented on src/main/java/Espresso.java:
```diff
@@ -0,0 +1,140 @@
+    void addToList(String str) {
+        Task task = null;
+        try {
+                if (str.startsWith("todo")) {
```
You may change the if-statement indentation to indentation to once (4 spaces) instead of twice.

ChinSekYi commented on src/main/java/deadlineTask.java:
You may rename your class to be capitalised at the first letter.

ChinSekYi commented on src/main/java/eventTask.java:
```diff
@@ -0,0 +1,14 @@
+public class eventTask extends Task{
+    private String starts,ends;
```
You may put a whitespace after the comma.
```suggestion
    private String starts, ends;
```

ChinSekYi reviewed (commented):
Overall, your code are of good quality, I did not find much code quality issues. I have only raise up a few suggestions upon reviewing your code.

KiKuasaurus commented:
Overall the code is well written without much issues, but perhaps javadoc could be added to make it easier to understand the purpose of each method and class.

Nihirraa committed:
Added TaskType Enum

Nihirraa committed:
Add Enums

Nihirraa committed:
Added saving functionality for tasks

Nihirraa committed:
Resolved merge conflict in src/main/java/TaskType.java

Nihirraa committed:
Add date formats

Nihirraa committed:
Add date formats

Nihirraa committed:
Merge branch 'branch-Level-8'

Nihirraa committed:
Added OOP properties to source code

Nihirraa committed:
Add gradle build files

Nihirraa committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Nihirraa committed:
Gradle build file

Nihirraa committed:
add packages

Nihirraa committed:
Added JUnit Testing

Nihirraa committed:
Add JavaDoc comments to classes and methods

Nihirraa committed:
Applied coding standards

Nihirraa committed:
Added finding function

Nihirraa committed:
Merge branch 'branch-A-CodingStandard'

Nihirraa committed:
Merge branch 'branch-Level-9'

Nihirraa committed:
Added GUI for my chatbot

Nihirraa committed:
Add changes for A-Assertions

Nihirraa committed:
Add changes for A-CodeQuality

Nihirraa committed:
Add changes for A-Streams

Nihirraa committed:
Add minute changes to code for better comprehension

Nihirraa committed:
Merge pull request #2 from Nihirraa/branch-A-Assertions

Add A-Assertions Feature

Nihirraa committed:
Merge pull request #3 from Nihirraa/branch-A-CodeQuality

Improve Code Quality

Nihirraa committed:
Merge pull request #4 from Nihirraa/branch-A-Streams

Add A-Streams Feature

Nihirraa committed:
Add assertions to MainWindow class

Assertions were added to document important assumptions in the Espresso instance, user input, and bot response. These checks enhance code robustness.

Nihirraa committed:
Ensure conflict detection between EventTask and DeadlineTask

- Enhanced conflict detection to prevent overlaps between EventTask and DeadlineTask.
- Conflict detection checks if a deadline occurs within the time span of an event.
- Throws an  when tasks with scheduling conflicts are detected.

Nihirraa committed:
Add more features to improve code quality

Nihirraa committed:
Add User Guide

Nihirraa committed:
Add Ui.png

Nihirraa committed:
Make final changes before releasing jar file

Nihirraa committed:
Update README.md

Nihirraa committed:
Fix README.md

Nihirraa committed:
Final changes for jar file
